src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java

```diff
@@ -24,32 +24,214 @@
+            MatcherAssert.assertThat(
+                ex.getMessage(),
+                Matchers.is("code is not provided")
+            );
+        }
+    }
+
+    /**
+     * Invalid user first name filed in json.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void invalidFirstNameField() throws IOException {
+        Mockito.when(
+            this.memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUserInvalidFirstName.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        // @checkstyle MultipleStringLiteralsCheck (1 line)
+        MatcherAssert.assertThat(
+            new PsLinkedin(this.memberprofilejson)
+                .enter(REQUEST).next()
+                .properties().get("first_name"),
+            Matchers.is("?")
+        );
+    }
+
+    /**
+     * Invalid user last name filed in json.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void invalidLastNameField() throws IOException {
+        Mockito.when(
+            this.memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUserInvalidLastName.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        MatcherAssert.assertThat(
+            new PsLinkedin(this.memberprofilejson)
+                .enter(REQUEST).next()
+                .properties().get("last_name"),
+            Matchers.is("?")
+        );
+    }
+
+    /**
+     * Invalid user ID filed in json.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void invalIdField() throws IOException {
+        Mockito.when(
+            this.memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUserInvalidId.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        try {
+            new PsLinkedin(this.memberprofilejson)
+                .enter(REQUEST);
+        } catch (final ClassCastException ex) {
+            MatcherAssert.assertThat(
```

Comment:
@ikhvostenkov this won't be checked if there is no exception, so this test might silently stop working